src/api/__tests__/apiHelpers.test.ts

```diff
@@ -241,4 +249,56 @@ describe('[api] helpers', () => {
       expect(result).toEqual({ error: 'Le refresh token est expiré' })
     })
   })
+
+  describe('handleGeneratedApiResponse', () => {
+    it('should return body without throwing error if status is 200', async () => {
+      const response = await respondWith('apiResponse')
+
+      const result = await handleGeneratedApiResponse(response)
+      expect(result).toEqual('apiResponse')
+    })
+
+    it('should return empty object without throwing error if status is 204 (no content)', async () => {
+      const response = await respondWith('', 204)
+
+      const result = await handleGeneratedApiResponse(response)
+      expect(result).toEqual({})
+    })
+
+    it.each([
+      400, // Bad Request
+      401, // Unauthorized
+      403, // Forbidden
+      404, // Not Found
+      500, // Internal Server Error
+      503, // Service Unavailable
+    ])('should throw error if response is not OK', async (statusCode) => {
+      const response = await respondWith('apiResponse', statusCode)
+      expect(response.ok).toBeFalsy()
+
+      const getResult = async () => {
+        await handleGeneratedApiResponse(response)
+      }
+      const error = new ApiError(
+        statusCode,
+        'apiResponse',
+        `Échec de la requête ${response.url}, code: ${response.status}`
+      )
+      await expect(getResult).rejects.toThrow(error)
+    })
+
+    it('should capture exception when access token is corrupted', async () => {
```

Comment:
je sais que `corrupted` est dans les commentaires, mais en vrai ce n'est pas vraiment corrompu, ça peut etre seulement expiré

```suggestion
    it('should navigate to login when access token is invalid', async () => {
```